### ISSUE TITLE
Use aria-label when computing name of hidden elements

### ISSUE THREAD

**Issue by jaragunde** [w3c/accname#144]
I'm reporting a case that I came up when implementing the proposal in PR #137, related to the discussion in #57, in Chromium.

When the hidden element which is a target of an aria-labelledby relation has the aria-label property, it might be preferable to use that instead of innerText. An example, based on [an existing Chromium test](https://chromium.googlesource.com/chromium/src/+/refs/heads/main/content/test/data/accessibility/aria/hidden-labelled-by.html):

```html
<html>
<style>
  .hide {
    visibility: hidden;
  }
  .visible {
    visibility: visible;
  }
</style>
<body>
  <div>
      <span role="group" class="visible" id="test" aria-labelledby="a4">
        <span class="hide" aria-label="span-A4" id="a4"></span>
      </span>
  </div>
</body>
</html>
```

The name of the node with id="test" used to be 'span-A4', but after implementing the  PR #137 proposal, the name is empty.

**Comment by cookiecrook** [comment]
Seems reasonable to special case `aria-label`.

What would you expect to happen in the case of an empty `aria-label`, rather than a missing attribute? Would this step result in "" or "contents"?

```html
<span hidden id="foo" aria-label="">contents</span>
```

**Comment by cookiecrook** [comment]
Something like this?
```diff
-<li>If the <code>current node</code> is <a class="termref">hidden</a>, compute the text alternative of the <code>current node</code> by returning {{HTMLElement/innerText}} of the node.</li>
+<li>If the <code>current node</code> is <a class="termref">hidden</a>:
+  <ol>
+    <li>If the <code>current node</code> has an <a class="pref">aria-label</a> attribute whose value, when trimmed of white space, is not the empty string, return the value of <a class="pref">aria-label</a>.</li>
+    <li>compute the text alternative of the <code>current node</code> by returning {{HTMLElement/innerText}} of the node</li>
+  </ol>
+</li>
```

**Comment by aleventhal** [comment]
@cookiecrook +1

This will reduce the chances that we will break existing content, and is reasonably simple and intuitive.

**Comment by cookiecrook** [comment]
Assuming @jaragunde and @accdc will like this change, so I've added it as a change suggestion to PR #137. Once the change is pulled in there, this issue #144 can be closed.

**Comment by cookiecrook** [comment]
Closing since that's now part of #137. Re-open if needed. Thanks @jaragunde and @aleventhal.

**Comment by jaragunde** [comment]
Thanks to you for shaping this suggestion into the proposed text!
